zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -26,11 +31,69 @@ pub fn founders_reward(height: Height, network: Network) -> Result<Amount<NonNeg
     }
 }
 
+/// Get the founders reward t-address for the specified block height as described in [protocol specification §7.8][7.8]
+///
+/// [7.8]: https://zips.z.cash/protocol/canopy.pdf#foundersreward
+pub fn founders_reward_address(height: Height, network: Network) -> Result<String, Error> {
+    let blossom_height = Blossom
+        .activation_height(network)
+        .expect("blossom activation height should be available");
+    let canopy_height = Canopy
+        .activation_height(network)
+        .expect("canopy activation height should be available");
+
+    if height >= canopy_height {
+        panic!("no address returned after canopy");
+    }
+
+    let mut adjusted_height = height;
+    if height >= blossom_height {
+        adjusted_height = Height(
+            blossom_height.0
+                + ((height.0 - blossom_height.0) / (BLOSSOM_POW_TARGET_SPACING_RATIO as u32)),
+        );
+    }
+
+    let address_index = 1 + (adjusted_height.0 / FOUNDER_ADDRESS_CHANGE_INTERVAL as u32);
+    Ok(FOUNDERS_REWARD_ADDRESSES_MAINNET[(address_index - 1) as usize].to_string())
+}
+
+/// Returns a list of outputs in `Transaction`, which have a script address equal to `String`.
+pub fn find_output_with_address(transaction: &Transaction, address: &str) -> Vec<Output> {
+    let calculated_addr: Address = address.parse().unwrap();
+
+    // For debugging
+    let mut output_number = 0;
+    println!("Looking for founders reward address in new coinbase transaction:");
+    for o in transaction.outputs().iter() {
+        let lock_script = o.lock_script.clone();
+
+        // For one of the coinbase outputs the address we get from the lock_script should be one of
+        // the ones in the FOUNDERS_REWARD_ADDRESSES_MAINNET array, not happening.
+        let output_address: Address = Address::from(lock_script);
```

Comment:
This is how i am converting the script from the output to a t-address to later compare. This is wrong and where the problem is.

Reply:
I was pointed to https://github.com/zcash/librustzcash/blob/b1c3f9d3f033db9519d45c69d1a8e51dd6b710a5/zcash_primitives/src/legacy.rs#L42-L66 by @str4d (thank you!) today. I didn't checked the details yet, specially if we already have this implemented, if not, we probably need to add it.

Reply:
Ok, i ported the function provided into zebra and i can confirm it works as expected. I can do the matching now(after cleaning the mess).

Reply:
Ok, so i added this at https://github.com/ZcashFoundation/zebra/pull/1170/commits/499435ff7c0cac5b6d6e79c17c87b26fc1c35e90

I currently have some messy code with the types that will need to be fixed. Basically, given a script hash alone(as `lock_script`, the one we get from transaction outputs) it is not possible to know to which network it belongs, so i had to create a new type `ScriptForNetwork` and implement `impl From<ScriptForNetwork> for Address` to do the conversion. This can be probably avoided by using existing `impl From<Script> for Address` if the network can be available inside there or by some other way.

Anyway, i think what was blocking is resolved so marking this as resolved.

Reply:
`impl From<Script> for Address {` assumes we're on mainnet, which is a bug. Zebra needs to support both mainnet and testnet.

So that impl should be deleted, and any code that uses it should be rewritten.
Let's do that in #1187?

In this PR, we need tests that make sure that `From<ScriptForNetwork> for Address` actually works.

Reply:
I fixed the bug and wrote the tests in #1191.

I've asked some of the other developers if we should use a struct or a tuple.